src/jdk.jdi/share/classes/com/sun/tools/jdi/TargetVM.java

```diff
@@ -123,8 +123,9 @@ public void run() {
                 byte b[] = connection.readPacket();
                 if (b.length == 0) {
                     done = true;
+                } else {
+                    p = Packet.fromByteArray(b);
```

Comment:
Just to clarify, the IOException here is not actually causing any problems. It is caught immediately below, and we've already set done = true, so we still end up shutting down the connection. Is that correct? I'm okay with the change since it is a bit cleaner.